paths.tex

```diff
@@ -368,31 +373,40 @@ \subsection{Consecutive Shortest Path Distance Algorithm}
 
 \subsection{Complete Shortest Path Distance Algorithm}
 
-\begin{figure}[H]
-\begin{lstlisting}[frame = single]
-  Input : Graph (V,E,T), dist[][][] entries set to Inf
-  Output : dist[][][]
-
-  for (t,src) in {T(e) | e in E } x V:
-    R := {(src,src)}
-    S := {}
-    interval[t][] = repeat null |V| times.
-    dist[t][src][src] = 0
-    while R != {}:
-      pred_e := pop R
-      for e in edge_neighborhood(pred_e) - union S R:
-        if (t in intersect(interval[pred_e],T(e))) :
-          dist[t][src][snd e] := min(dist[t][src][fst e] + 1,
-                                        dist[t][src][snd e])
-          interval[t][e] := intersect(interval[pred_e],T(e))
-          insert(e,R)
-      S += pred_e
-  return dist
-\end{lstlisting}
-\caption{An algorithm to compute the lengths of all complete temporal shortest
-paths}
+\begin{algorithm}[H]
+\SetAlgoLined
+\KwIn{A temporal graph $G = (V, E, T)$ and a 3D array $dist$ which encapsulates
+  a complete mapping between pairs of vertices along with an initial timestamp
+  to the distance between them, with the default values set to $\infty$.}
+\KwOut{The algorithm outputs the 3D array $dist$ such that
+  $dist[t_i][src][dest] = d$ where $d$ is the (shortest complete path)
+  distance from node $src$ to node $dest$ such that time $t_i$ intersects the time
+  interval of the first edge in the path.}
+  \ForEach(//loop over nodes in time slices){
+    start node $(t, v_{src})$ in $V \times \mathcal{T}$
+  }{
+    Let the reached edges min-heap $R = \{(v_{src}, v_{src})\}$\;
+    Let the searched edges $S = \varnothing$\;
+    Let $dist[t][src][src] = 0$\;
+    \For{$i$ in $range(|V|)$}{
+      Let $interval[t][i] = null$\;
```

Comment:
can also pass `interval` as a parameter with the precondition that it is a complete map with defaults at null, removing the ugly (and not algorithmically interesting) for-loop.  Doing this seems more of an implementation detail than a algorithm detail